File: src/main/java/org/example/collection/validators/ValidatorPersonPasportID.java
package org.example.collection.validators;

import org.example.exception.InvalidDataError;
/**
 * Валидатор номера паспорта человека
 */

public class ValidatorPersonPasportID implements Validator<String> {
    /**
     * Проверяет строку на пустоту
     * @param s строка
     * @throws InvalidDataError если строка пуста или меньше 6 символов
     */
    @Override
    public void valide(String s) throws InvalidDataError {
        if (s.isEmpty() | s.length()<6){
            throw  new InvalidDataError();
        }
    }
}

File: src/main/kotlin/collection/validators/ValidatorPersonPasportID.kt
package org.chousik.collection.validators

import org.chousik.exception.InvalidDataError


class ValidatorPersonPasportID : IValidator<String?> {

    override fun valide(s: String?) {
        if (s!!.isEmpty() or (s.length < 6)) {
            throw InvalidDataError()
        }
    }
}